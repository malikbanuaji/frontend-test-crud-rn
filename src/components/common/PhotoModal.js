import React from 'react';
import {StyleSheet, Text, View, TouchableNativeFeedback} from 'react-native';
import Modal from 'react-native-modal';
import {Colors, Spacing, Typography, Mixins} from '../../styles';

export default function PhotoModal({
  onPressDelete,
  onPressCamera,
  onPressGallery,
  onPressAbort,
  hasPhoto,
  isVisible,
}) {
  return (
    <Modal
      testID={'pick-photo-modal'}
      onBackdropPress={onPressAbort}
      onBackButtonPress={onPressAbort}
      hideModalContentWhileAnimating={false}
      animationIn={'fadeIn'}
      animationOut={'fadeOut'}
      useNativeDriver={false}
      backdropOpacity={0.6}
      backdropTransitionOutTiming={0}
      isVisible={isVisible}>
      <View style={styles.container}>
        <Text style={styles.title}>Ganti Foto</Text>
        <View style={styles.titleUnderline} />
        {hasPhoto && (
          <TouchableNativeFeedback
            testID={'photo-modal-delete-photo'}
            onPress={onPressDelete}>
            <View style={styles.button}>
              <Text style={styles.buttonItemText}>Hapus foto</Text>
            </View>
          </TouchableNativeFeedback>
        )}
        <TouchableNativeFeedback
          testID={'photo-modal-take-photo'}
          onPress={onPressCamera}>
          <View style={styles.button}>
            <Text style={styles.buttonItemText}>Ambil foto baru</Text>
          </View>
        </TouchableNativeFeedback>
        <TouchableNativeFeedback
          testID={'photo-modal-select-gallery'}
          onPress={onPressGallery}>
          <View style={styles.button}>
            <Text style={styles.buttonItemText}>Pilih foto baru</Text>
          </View>
        </TouchableNativeFeedback>
        <View style={styles.cancelButtonContainer}>
          <View style={{...Mixins.buttonRounded}}>
            <TouchableNativeFeedback
              testID={'photo-modal-abort'}
              onPress={onPressAbort}>
              <View style={styles.cancelButtonView}>
                <Text style={styles.actionButtonText}>Batal</Text>
              </View>
            </TouchableNativeFeedback>
          </View>
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.white,
    padding: Spacing.base,
    borderRadius: Mixins.borderRadius,
  },
  title: {...Typography.modalTitle, padding: Spacing.small},
  titleUnderline: {
    height: 1.5,
    backgroundColor: Colors.lightGray,
  },
  button: {
    padding: Spacing.small,
  },
  cancelButtonContainer: {
    alignSelf: 'flex-end',
  },
  cancelButtonView: {
    padding: Spacing.smaller,
  },
  actionButtonText: {
    ...Typography.modalButtonActionText,
  },
  buttonItemText: {
    ...Typography.modalButtonItemText,
  },
});
