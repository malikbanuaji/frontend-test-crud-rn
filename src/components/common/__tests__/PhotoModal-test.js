import React from 'react';
import renderer from 'react-test-renderer';
import PhotoModal from '../PhotoModal';

test('renders correctly without photo', () => {
  const tree = renderer.create(<PhotoModal hasPhoto={false} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly with photo', () => {
  const tree = renderer.create(<PhotoModal hasPhoto={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
