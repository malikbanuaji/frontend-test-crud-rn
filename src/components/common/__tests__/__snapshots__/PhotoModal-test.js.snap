// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with photo 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 20,
            "padding": 12,
          }
        }
      >
        Ganti Foto
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#d3d3d3",
            "height": 1.5,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Regular",
              "fontSize": 14,
            }
          }
        >
          Hapus foto
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Regular",
              "fontSize": 14,
            }
          }
        >
          Ambil foto baru
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Regular",
              "fontSize": 14,
            }
          }
        >
          Pilih foto baru
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 4,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                }
              }
            >
              Batal
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly without photo 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    forwardedRef={[Function]}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "Rubik-Light",
            "fontSize": 20,
            "padding": 12,
          }
        }
      >
        Ganti Foto
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "#d3d3d3",
            "height": 1.5,
          }
        }
      />
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Regular",
              "fontSize": 14,
            }
          }
        >
          Ambil foto baru
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "padding": 12,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Rubik-Regular",
              "fontSize": 14,
            }
          }
        >
          Pilih foto baru
        </Text>
      </View>
      <View
        style={
          Object {
            "alignSelf": "flex-end",
          }
        }
      >
        <View
          style={
            Object {
              "borderRadius": 4,
              "overflow": "hidden",
            }
          }
        >
          <View
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "padding": 8,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                }
              }
            >
              Batal
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
